File: tests/integration/benchmarks/fixtures/csharp/rosetta/src/Models/ModelFactory.cs
using Rosetta.Types;

namespace Rosetta.Models;

/// <summary>
/// Factory and validation utilities for domain models.
/// </summary>
public static class ModelFactory
{
    /// <summary>
    /// Factory for creating records with sensible defaults.
    /// </summary>
    public static Record CreateRecord(string id, double value) => new(id, value);

    /// <summary>
    /// Validates configuration is within acceptable bounds.
    /// </summary>
    public static bool ValidateConfig(ProcessorConfig config) =>
        config.BatchSize > 0 && 
        config.Timeout > 0 && 
        config.BatchSize <= 1000;
}

File: tests/integration/benchmarks/fixtures/java/rosetta/src/com/rosetta/models/ModelFactory.java
package com.rosetta.models;

import com.rosetta.types.ProcessorConfig;

/**
 * Factory and validation utilities for domain models.
 */
public final class ModelFactory {
    private ModelFactory() {}

    /**
     * Factory for creating records with sensible defaults.
     */
    public static Record createRecord(String id, double value) {
        return new Record(id, value);
    }

    /**
     * Validates configuration is within acceptable bounds.
     */
    public static boolean validateConfig(ProcessorConfig config) {
        return config.getBatchSize() > 0 &&
               config.getTimeout() > 0 &&
               config.getBatchSize() <= 1000;
    }
}
